### PR TITLE
refactor: hoist collections data

### PR DESCRIPTION
**Pull request by UncleBill**
MF-0000

improve handling blocked collectibles

### CODE REVIEW

**Comment by UncleBill** [comment]
@guanbinrui PTAL

File: packages/shared/src/UI/components/AssetsManagement/UserAssetsContext.tsx [hunk 1]
```diff
@@ -1,200 +1,32 @@
...
-                }
-            }
-            const iterator = assetsLoaderIterators.current.get(stateKey) || generate(collection)
-            assetsLoaderIterators.current.set(stateKey, iterator)
-            const result = await iterator.next()
-            if (result.done) dispatch({ type: 'SET_LOADING_STATUS', id: stateKey, finished: true, loading: false })
-        },
-        [Hub, address, loadAssetsViaHub],
+    const blockedIds = useMemo(() => {
+        if (systemAccount !== account) return EMPTY_LIST
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/AssetsManagement/UserAssetsContext.tsx]: isSameAddress

File: packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx [hunk 2]
```diff
@@ -207,27 +172,16 @@ export const CollectionList = memo(function CollectionList({
...
-                await delay(30)
-                // The first collectables might have been blocked, if assets of
-                // current collection is empty, try loading another page to fill
-                // the blank.
-                if (!assetsMapRef.current[stateKey].assets.length) {
-                    await load()
-                }
-            }
-            load()
+            loadAssets(collection)
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx:184]: Lost await?

Comment (UncleBill) [reply packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx:184]: It's unnecessary.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**